internal/zinc-compile-core/src/test/scala/sbt/internal/inc/javac/JavaCompilerSpec.scala

```diff
@@ -95,7 +95,7 @@ class JavaCompilerSpec extends UnitSpec {
         val (result, _) = compile(local, Seq(input), Seq("-d", out.getAbsolutePath))
         result shouldBe true
         val clazzz = new URLClassLoader(Array(out.toURI.toURL)).loadClass("hasstaticfinal")
-        ClassToAPI(Seq(clazzz))
+        ClassToAPI(Seq(clazzz)).sortBy(_.definitionType.hashCode)
```

Comment:
1. There is no issue with Contraband. Changing how `hashCode()` is generated based on the laziness of the fields is intended and prevents infinite recursion (see the test added in 436828c).
  2. The compilation done by `compileWithPrimitive()` can produce a *class* and an *object*, but their order is not defined since they're in a `Set`. So `leftApi` will contain a *class* and an *object* from a first compilation and `rightApi` will contain a *class* and an *object* from a second compilation. Since we want to compare the *class* of `leftApi` with the *class* of `rightApi` (same for the *object*) we need to form the correct pairs for the comparison and not compare a *class* with an *object*. Sorting ensure we have the same order in `leftApi` and in `rightApi`. There may be other ways to ensure we get the correct pairs, the sort could be done elsewhere, but sorting there seemed to be a simple fix for the problem. Finally the sort is done only in the tests, it shouldn't be needed elsewhere.

Sorry for the late response. Even though the PR has been merged, I hope these answers help you understand the necessity of the sort in those tests.